### ISSUE TITLE
User can create saved searches but only admin can delete them

### ISSUE THREAD

**Issue by jalogisch** [Graylog2/graylog2-server#8918]
## Expected Behavior
As a normal User you should be able to delete a saved search that you have previous created.

## Current Behavior
When delete a previous created view, the buttom notification banner first is green showing "deleted proper" and red "no access" in the same second and you get the monkey with the message:

```
The permissions check for the following request failed,
while trying to access /search.
cannot DELETE https://graylog.z.jalogis.ch/api/views/5f522696342d5f696ba96b93 (403)
````

![Screen-Recording-2020-09-04-at-1](https://user-images.githubusercontent.com/404238/92239190-e023ed00-eeba-11ea-98b3-defc4994e77b.gif)

## Steps to Reproduce (for bugs)
1. Create a User with `Reader` role and access to a search
2. login with that user
3. Create a saved search
4. try to delete 

## Context
As Admin you are not able to deny that people create saved searches but they will not be able to housekeep this and you have log of fragments left. 

## Your Environment
* Graylog Version: 3.3.4

**Comment by jalogisch** [comment]
Same is true for the dashboards, as later investigation showed:

https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/ViewsResource.java#L206

That need to be fixed too.

**Comment by dennisoelkers** [comment]
Fixed in #9067, #9101.
